### PR TITLE
fix: options that were silently dropped, and runs that reported the wrong outcome

### PR DESCRIPTION
**Pull request by NevermindNilas**
Options the user explicitly asked for were being dropped without a word, and several runs reported the wrong outcome. This branch fixes both classes, plus the BT.2020 and container bugs found while testing them.

## Behaviour change

An `*_nelux` `--encode_method` combined with `--custom_encoder`, `--bit_depth 16bit`, `--output_scale` or `--depth` now runs the equivalent plain encoder instead, with a warning. Those runs lose the in-process encoder's speed and gain an output that matches what was asked for.

## Silently dropped options

- `--encode_method *_nelux` ignored `--custom_encoder`, `--bit_depth 16bit` and `--output_scale`. `--output_scale 320x180` wrote a 1280x720 file.
- `--output_scale` auto-enabled `--resize`, which applied `--resize_factor`'s default of 2 to the decode. `--upscale --output_scale 1280x720` on a 640x360 source ran the model over 1280x720 to 2560x1440 and then scaled back down. The model now sees 4x fewer pixels for the same output.
- `--depth` encoded through OpenCV with a hardcoded `mp4v` fourcc on 25 of the `--depth_method` choices (every `og_*` model and the whole `small_v3`/`base_v3`/`large_v3` family), discarding `--encode_method`, `--custom_encoder` and `--bit_depth`. They now use the shared writer, so `--bit_depth 16bit` gives a 10-bit depth map. Depth values are unchanged.
- `--bit_depth 16bit` on `nvenc_h265`, `slow_nvenc_h265`, `nvenc_av1` and `slow_nvenc_av1` uploaded 8-bit `nv12` surfaces, so 16-bit depth maps were written 8-bit. They now upload `p010le`.
- `--output` with a `.webm` extension kept the default `x264` because the encoder check read the *input* path. FFmpeg refused the header and left a 262-byte file the run called a success.
- A `--json` config auto-enabled capabilities it had itself switched off. The After Effects panel serializes its whole form, so every capability's `*_method` arrived at its default and counted as "the user asked for this". An explicit `"upscale": false` also lost to its own sibling `"upscale_method"`.

## Runs that reported the wrong outcome

- `--encode_method png` and `--autoclip` printed "Processing FAILED" and exited 1 on every successful run, and reported failure to After Effects. Both now exit 0; a genuinely empty sequence folder still fails.
- A `--depth` inference error was logged and ignored: short file, exit 0, success reported to AE. It now exits 1, and a frame loop that raises no longer hangs the run.
- A `--moblur` worker thread that died took its exception into a discarded future; a failure in its setup hung the run; a 0-byte output was reported to AE as completed.
- A decode that died part-way through finished as a shorter "successful" video, on every capability. It now exits 1, while a decoder error raised after the last frame still exits 0 because that output is complete.
- `--segment` and `--obj_detect` swallowed per-frame errors the same way, so a run that dropped every frame reported success.
- Errors raised outside `start()` (unreadable metadata, a download out of retries, a validator exit) left the AE panel on the last progress string. `--depth` never sent a terminal status at all. Every run now sends exactly one.

## Colour and container

- Any BT.2020/HDR10 source failed its entire render. The filter asked zscale for `matrix=bt2020:norm=bt2020` and neither token exists, so FFmpeg rejected the filtergraph for a 0-byte output. The arm now uses `matrix=bt2020nc`, tags the output with the source's own transfer so PQ, HLG and SDR BT.2020 each keep their curve, and works at 16-bit like the BT.709 arm. A 16-bit ramp went from 220 distinct 10-bit codes, every one a multiple of 4, to a full ladder. A source carrying the BT.2020 matrix but no primaries is detected now instead of being converted to BT.709.
- An `.mp4`/`.m4v` output stream-copied text subtitles, so an ordinary anime MKV aborted the mux at header-write and lost the whole render after the models had loaded. Those now transcode to `mov_text`, while subtitles the container already accepts (`mov_text`, `ttml`) are still copied. Extensions match case-insensitively.
- `--segment` pushed RGBA through `yuva420p` before the `yuva444p10le` ProRes 4444 output, discarding half the chroma resolution and two bits ahead of a conversion that cannot recover either. On a 1-pixel chroma pattern the old chain measured 7.3 dB against the source, the new one 48.5 dB.
- Image-sequence input was completely dead: the metadata probe rejected every `frames_%05d.png` pattern as a missing file. `--input <folder of PNGs>` works again.

## Models and dedup

- `--depth_method og_distill_small_v2` and `og_distill_base_v2` resolved `og_small_v2`'s and `og_base_v2`'s weights, so they produced byte-identical output under a name promising a different model. They now load the real Distill-Any-Depth checkpoints.
- `--depth_method og_distill_large_v2` (and `-mps`) removed: a Distill-Any-Depth model built against a Depth-Anything-V2 checkpoint, it always crashed. Its architecture has no reference implementation, so `distill_large_v2` is the same model.
- `--dedup_method flownets` advanced its reference frame on duplicates, so a slow pan or fade was deleted frame by frame. On a 96-frame fade it kept 52 frames; it now keeps 68.
- `--dedup_method flownets` and `vmaf-cuda` crashed at model init on macOS and CUDA-less Linux with an error that never named the flag. They now fall back like `--smooth_dedup` already did.
- `--interpolate <rife>-directml` set torch's global default dtype to float16 and never restored it, so in a batch run every later stage, including videos 2..N, allocated in fp16.
- `--smooth_dedup` kept a whole widened gap in memory at once: 5.1 GB VRAM after a 2-second 1080p hold, 10.9 GB after 12 seconds, against a flat 4.5 GB for plain `--interpolate`. `--smooth_dedup_max_span` never bounded it. Now flat at 4.4 GB with byte-identical output.

## Docs

- `PARAMETERS.MD` documents `--decode_method`, `--profile`, `--obj_detect_disable_annotations` and the six `*_nelux` encoders. Its debug section told you to run `--preset_list`, and two examples used `--bit_depth 10bit`; neither has ever been accepted.
- `--precision` removed. It was never implemented and never read; `--half` is the real knob.

## Unrelated, but it was blocking this PR

`src/masking.py`'s self-referential `MaskedSink` return annotation only survives at runtime because Python 3.14 defers annotation evaluation. `ty` reads it as a first-party name that does not exist and fails the blocking lint job, red on `main` since 2026-08-03. Quoting it trips ruff's UP037, so it is now `typing.Self`, which both accept. No behaviour change.

## Tests

New: `tests/test_docsDrift.py` (a CLI flag or `--encode_method` choice missing from `PARAMETERS.MD`, a documented flag the parser rejects, or an example using a value outside a flag's `choices`), `tests/test_depthWriters.py` (a depth backend encoding with OpenCV instead of the shared writer), `tests/test_runOutcome.py` (a standalone capability that stops reporting its own outcome), `tests/test_ffmpeg_writer.py`, `tests/test_colorspaceFilters.py`. Extended: `test_argumentsChecker.py`, `test_batchFailure.py`, `test_ffmpegColorspace.py`, `test_registryDrift.py`.

### CODE REVIEW

File: src/io/runOutcome.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        return False
+    for name in entries:
+        if not name.startswith(prefix) or not name.endswith(suffix):
+            continue
+        try:
+            if os.path.getsize(os.path.join(directory, name)) > 0:
+                return True
+        except OSError:
+            continue
+    return False
```
[the remark this answers is not in the document]
Comment (NevermindNilas) [reply src/io/runOutcome.py:60]: Confirmed and fixed in ddf8aa36. `sequenceWrote()` now matches the `%d`/`%0Nd` counter itself rather than the prefix and extension, so a stray `frames_old.png` no longer counts as a written frame. The padded width is treated as a minimum, not a fixed length, because FFmpeg writes a sixth digit once a `%05d` counter passes 99999; `%d` needs at least one digit.

The line 75 half (`outputWasWritten()` taking the sequence branch for any `%` in the basename) is fixed separately in 75a0ac06 — it now requires the same counter, so an ordinary `50%_off.mp4` is judged by its own size again.

Covered by four new cases in `tests/test_runOutcome.py`: a stray `frames_old.png`/`frames_.png`, a counter past the padded width, an unpadded `%d`, and a literal percent in a file name. Full suite: 671 passed.

File: main.py [hunk 2]
```diff
@@ -517,8 +568,40 @@ def _interpolateOrHold(self, frame: any, sink: any, nextFrame: any) -> None:
...
+        if not self.interpQueue.frames:
+            return
+
+        import torch
+
+        if torch.cuda.is_available():
+            with torch.cuda.stream(torch.cuda.default_stream()):
+                self._writeInterpFrames()
+        else:
+            self._writeInterpFrames()
```
[the remark this answers is not in the document]
Comment (NevermindNilas) [reply main.py:595]: Checked this one against every backend that puts into the queue, and I do not think the wait is needed — leaving it as is.

The hazard would be real if a driver could `put()` a tensor whose producing stream still had work in flight. None of them can: each host-synchronizes the producing stream immediately before the put.

- `src/gmfss/gmfss.py:198-202` — `self.stream.synchronize()`, then the `interpQueue.put(output)` loop. This is the one case where a private stream is still *current* at put time (`stream_context()` wraps the whole `__call__`), which is exactly what the default-stream pinning is for.
- `src/interpolate/rife.py:543-544` — `processFrame(..., "infer")` ends in `normStream.synchronize()`, and the put sits at `__call__` scope with no stream context current.
- `src/interpolate/rife_tensorrt.py:434-438` — `stream.synchronize()`, `outputStream.synchronize()`, then the put, dedented out of both `with` blocks.
- `src/interpolate/distildrba.py:242-255` and `:604-605` — same shape, synchronize then put.

`Stream.synchronize()` blocks the host until that stream's work has completed, which is strictly stronger than `default_stream().wait_stream(producer)`: once it returns the writes are finished and visible to every stream, so an event-based wait afterwards would record against an already-idle stream and be a no-op.

That ordering is not incidental — it is the `WriteBuffer` handoff contract this repo already depends on. The writer copies on its own private stream and never waits on the caller's, so a producer that queued work after its sync would corrupt output through the normal `_writeOut` path too, long before the mid-gap drain saw it.

So the pinning here is about *which stream the subsequent upscale and write enqueue on*, not about data visibility: without it a mid-gap drain fired from inside gmfss's stream context would hand the writer work on a private stream and break that contract. The docstring at `main.py:578-584` says this; I have reworded it in c806c87a is not accurate — no code change was made for this thread.

[the remark this answers is not in the document]
Comment (NevermindNilas) [reply main.py:595]: Correction to the last line of the previous reply, which garbled itself: no code change was made for this thread. The reasoning stands as written — the docstring at `main.py:578-584` already records why the drain is pinned to the default stream.
